### PR TITLE
Use funding cycle to determine project version

### PR DESCRIPTION
**Pull request by tomquirk**
## What does this PR do and why?

_Provide a description of what this PR does. Link to any relevant GitHub issues, Notion tasks or Discord discussions._

## Screenshots or screen recordings

_If applicable, provide screenshots or screen recordings to demonstrate the changes._

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [ ] I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

File: src/utils/server/pages/props.ts [hunk 1]
```diff
@@ -58,19 +59,24 @@ export async function getProjectProps(
...
+      hasFundingCycle(projectId, CV_V2),
+      hasFundingCycle(projectId, CV_V3),
     ])
 
-    const cv = isV3ProjectResult ? CV_V3 : CV_V2
+    const initialCv = hasV3FundingCycle ? CV_V3 : CV_V2
+
+    const cvs: CV2V3[] = []
+    if (hasV2FundingCycle) cvs.push(CV_V2)
+    if (hasV3FundingCycle) cvs.push(CV_V3)
```
Comment (tomquirk) [inline src/utils/server/pages/props.ts:72]: We'll use this to determine if we should show the version selector, or just the static badge.

File: src/utils/server/pages/props.ts [hunk 2]
```diff
@@ -58,19 +59,24 @@ export async function getProjectProps(
...
-    const [metadata, isV3ProjectResult] = await Promise.all([
+    const [metadata, hasV2FundingCycle, hasV3FundingCycle] = await Promise.all([
       findProjectMetadata({ metadataCid }),
-      isV3Project(projectId),
+      hasFundingCycle(projectId, CV_V2),
+      hasFundingCycle(projectId, CV_V3),
     ])
 
-    const cv = isV3ProjectResult ? CV_V3 : CV_V2
+    const initialCv = hasV3FundingCycle ? CV_V3 : CV_V2
```
Comment (tomquirk) [inline src/utils/server/pages/props.ts:68]: If they have a v3 fc, load project page using v3 contracts. Else use V2

File: src/utils/server/pages/props.ts [hunk 3]
```diff
@@ -58,19 +59,24 @@ export async function getProjectProps(
 
   try {
     const metadataCid = await getMetadataCidFromContract(projectId)
-
-    const [metadata, isV3ProjectResult] = await Promise.all([
+    const [metadata, hasV2FundingCycle, hasV3FundingCycle] = await Promise.all([
       findProjectMetadata({ metadataCid }),
-      isV3Project(projectId),
+      hasFundingCycle(projectId, CV_V2),
+      hasFundingCycle(projectId, CV_V3),
```
Comment (tomquirk) [inline src/utils/server/pages/props.ts:57]: Check if a project has an FC on V2, and on V3.

File: src/utils/server/pages/props.ts (hunk 1 above)
Comment (peripheralist) [reply src/utils/server/pages/props.ts:64]: we show the selector if there's both v2 and v3 activity ya?

File: src/components/v2v3/V2V3Project/V2V3ProjectHeaderActions/ContractVersionSelect.tsx [hunk 4]
```diff
@@ -6,22 +6,22 @@ import { V2V3ContractsContext } from 'contexts/v2v3/V2V3ContractsContext'
...
-  const { setVersion, cv } = useContext(V2V3ContractsContext)
+  const { setVersion, cv, cvs } = useContext(V2V3ContractsContext)
+
+  const SELECT_OPTIONS: BaseOptionType[] =
+    cvs?.map(cv => ({
+      label: CV_LABELS[cv],
+      value: cv,
+    })) ?? []
 
   if (SELECT_OPTIONS.length < 2) {
```
Comment (peripheralist) [inline src/components/v2v3/V2V3Project/V2V3ProjectHeaderActions/ContractVersionSelect.tsx:26]: 👌

File: src/utils/server/pages/props.ts (hunk 1 above)
Comment (tomquirk) [reply src/utils/server/pages/props.ts:64]: :getsit:

**Pull request merged**